pygpudrive/env/wrappers/sb3_wrapper.py

```diff
@@ -198,6 +208,46 @@ def seed(self, seed=None):
         self._seeds = [seed + idx for idx in range(self.num_envs)]
         return self._seeds
 
+    def resample_scenario_batch(self):
+        """Swap out the dataset."""
+        if self.exp_config.resample_mode == "random":
+            total_unique = len(self.unique_scene_paths)
+
+            # Check if N is greater than the number of unique scenes
+            if self.num_worlds <= total_unique:
+                dataset = random.sample(
+                    self.unique_scene_paths, self.num_worlds
+                )
+
+            # If N is greater, repeat the unique scenes until we get N scenes
+            dataset = []
+            while len(dataset) < self.num_worlds:
+                dataset.extend(
+                    random.sample(self.unique_scene_paths, total_unique)
+                )
+                if len(dataset) > self.num_worlds:
+                    dataset = dataset[
+                        : self.num_worlds
+                    ]  # Trim the result to N scenes
+        else:
+            raise NotImplementedError(
+                f"Resample mode {self.exp_config.resample_mode} is currently not supported."
+            )
+
+        # Re-initialize the simulator with the new dataset
+        print(
+            f"Re-initializing sim with {len(set(dataset))} {self.exp_config.resample_mode} unique scenes.\n"
+        )
+        self._env.reinit_scenarios(dataset)
+
+        # Update controlled agent mask
+        self.controlled_agent_mask = self._env.cont_agent_mask.clone()
+        self.max_agent_count = self.controlled_agent_mask.shape[1]
+        self.num_valid_controlled_agents_across_worlds = (
+            self.controlled_agent_mask.sum().item()
+        )
+        self.num_envs = self.controlled_agent_mask.sum().item()
```

Comment:
This code seems duplicated across reinit_scenarios?

Reply:
Yes, removed redundancy!